File: output/GPT3.5-Turbo/xCodeEval_4000/C++/C/f4841462a97bd925085b8110830ca3f4.c
#include <stdio.h>
#include <stdbool.h>
#include <math.h>

#define CHUNK_SIZE 100000

bool prime(int N)
{
    if(N < 2) return false;
    for(int i = 2; i * i <= N; i++)
    {
        if(N % i == 0) return false;
    }
    return true;
}

bool sum2sq(int N)
{
    if(N != 2 && N % 4 != 1) return false;
    for(int A = 1; A * A < N; A++)
    {
        int B = (int)max(sqrt((double)(N - A * A)) - 2.0, 0.0);
        while(A * A + B * B < N) B++;
        if(A * A + B * B == N) return true;
    }
    return false;
}

int solve(int N)
{
    int baseInd = N / CHUNK_SIZE;
    int result = A[baseInd];
    for(int i = baseInd * CHUNK_SIZE + 1; i <= N; i++)
    {
        result += (prime(i) && sum2sq(i));
    }
    return result;
}

int main()
{
    int l, r;
    scanf("%d %d", &l, &r);
    printf("%d", solve(r) - solve(l - 1));

    return 0;
}

File: output/GPT4o-Mini/xCodeEval_4000/C++/Python/f4841462a97bd925085b8110830ca3f4.py
import math

CHUNK_SIZE = 100000

def _fill():
    N = 300000005
    p = [True] * N
    p[0] = p[1] = False

    bound = 20005
    for i in range(2, bound):
        if p[i]:
            for j in range(i * i, N, i):
                p[j] = False

    result = 0
    i = 0

    with open("output.txt", "w") as out:
        while i < N:
            if i % CHUNK_SIZE == 0:
                out.write(f"{result}, ")
            if i % (CHUNK_SIZE * 100) == 0:
                out.write('\n')
            i += 1
            if i < N and (i == 2 or i % 4 == 1):
                result += p[i]

def prime(N):
    if N < 2:
        return False
    for i in range(2, int(math.sqrt(N)) + 1):
        if N % i == 0:
            return False
    return True

def sum2sq(N):
    if N != 2 and N % 4 != 1:
        return False
    for A in range(1, int(math.sqrt(N)) + 1):
        B = max(int(math.sqrt(N - A * A)) - 2, 0)
        while A * A + B * B < N:
            B += 1
        if A * A + B * B == N:
            return True
    return False

def solve(N):
    baseInd = N // CHUNK_SIZE
    result = A[baseInd]
    for i in range(baseInd * CHUNK_SIZE + 1, N + 1):
        result += (prime(i) and sum2sq(i))
    return result

if __name__ == "__main__":
    _fill()
    l, r = map(int, input().split())
    print(solve(r) - solve(l - 1))
